registry/dynamodb.go

```diff
@@ -103,20 +126,72 @@ func (im *DynamoDBRegistry) Records(ctx context.Context) ([]*endpoint.Endpoint,
 
 	orphanedLabels := sets.KeySet(im.labels)
 	endpoints := make([]*endpoint.Endpoint, 0, len(records))
+	labelMap := map[string]endpoint.Labels{}
+	txtRecordsMap := map[string]*endpoint.Endpoint{}
 	for _, record := range records {
 		key := record.Key()
 		if labels := im.labels[key]; labels != nil {
 			record.Labels = labels
 			orphanedLabels.Delete(key)
 		} else {
 			record.Labels = endpoint.NewLabels()
+
+			if record.RecordType == endpoint.RecordTypeTXT {
+				// We simply assume that TXT records for the TXT registry will always have only one target.
+				if labels, err := endpoint.NewLabelsFromString(record.Targets[0], im.txtEncryptAESKey); err == nil {
+					endpointName, isAAAA := im.mapper.toEndpointName(record.DNSName)
+					key := fmt.Sprintf("%s::%s", endpointName, record.SetIdentifier)
+					if isAAAA {
+						key += keySuffixAAAA
+					}
+					labelMap[key] = labels
+					txtRecordsMap[key] = record
+					continue
+				}
+			}
 		}
 
 		endpoints = append(endpoints, record)
 	}
 
 	im.orphanedLabels = orphanedLabels
 
+	// Migrate label data from TXT registry.
+	if len(labelMap) > 0 {
+		for _, ep := range endpoints {
+			if _, ok := im.labels[ep.Key()]; ok {
+				continue
+			}
+
+			dnsNameSplit := strings.Split(ep.DNSName, ".")
+			// If specified, replace a leading asterisk in the generated txt record name with some other string
+			if im.wildcardReplacement != "" && dnsNameSplit[0] == "*" {
+				dnsNameSplit[0] = im.wildcardReplacement
+			}
+			dnsName := strings.Join(dnsNameSplit, ".")
+			key := fmt.Sprintf("%s::%s", dnsName, ep.SetIdentifier)
+			if ep.RecordType == endpoint.RecordTypeAAAA {
+				key += keySuffixAAAA
+			}
+			if labels, ok := labelMap[key]; ok {
+				for k, v := range labels {
+					ep.Labels[k] = v
+				}
+				ep.SetProviderSpecificProperty(dynamodbAttributeMigrate, "true")
+				delete(txtRecordsMap, key)
+			}
+		}
+	}
+
+	// Remove any unused TXT ownership records owned by us
+	if len(txtRecordsMap) > 0 && !plan.IsManagedRecord(endpoint.RecordTypeTXT, im.managedRecordTypes) {
+		log.Infof("Old TXT ownership records will not be deleted because \"TXT\" is not in the set of managed record types.")
```

Comment:
maybe add %d and pass len(txtRecordsMap) ?

Reply:
What is the value to the admin of the count of legacy TXT ownership records?

This log line is to let the admin know why the old TXT records aren't being deleted like they probably expect and give them a clue as to what they should do to fix the problem.

Reply:
If the log shows that the number decreases, but needs long time because of rate limits or so, then you can just wait. If the number does not decrease they should do something.

Reply:
The number is not going to decrease. In the situation where this is logged, the old TXT ownership records are not going to be deleted because of the configuration issue that the log line is reporting.